### DISCUSSION TITLE
Повторяем изученное

### DISCUSSION THREAD

**Discussion by pingvin1308** [cleannetcode/CsharpForBeginers#11] [Матерые]
Домашка: повторите задачи, которые я выполнил на "Повторяем изученное #1" и "Повторяем изученное #2".
Если вкратце, то я поправил юнит тесты, переделал работу с конфигой для передачи секрета. Вывел реквест и респонз модели, добавил валидацию. Также реализовал репозиторий с помощью ef core.

**Comment by artyomorel** [comment]
Повтори два последних урока
https://github.com/artyomorel/Timesheet/tree/homework-13-repeat
Если не сложно можешь предложить какие фичи можно самостоятельно попробовать реализовать , а ты бы потом оценил

**Reply by pingvin1308** [reply]
По моему проект с задачами открыт, по этому можешь взяться за любую задачу и попробовать ее сделать)
Буду рад посмотреть)

**Comment by artyomorel** [comment]
1-Я прошлый раз делал один контроллер для всех работников ,но в этот раз исправил ошибку и использовал Enum
2-Починил баг по добавлению сотрудником более 24 часов в сутки 
https://github.com/artyomorel/Timesheet/tree/Additional-Task
Кстати Ром заметил ,что если используешь FluentValidation то проверку Model.State можно не делать ,он сам проверяет и если не проходит валидацию выкидывает чего не хватает. Завтра попробую с интеграцией с гитом поработать

**Reply by pingvin1308** [reply]
Неплохая реализация, нужно только чуток правок сделать и будет совсем хорошо)
Попрактикуешь написание юнит тестов)

**Reply by artyomorel** [reply]
Понял 
Сегодня все доработаю , огромное спасибо

**Comment by artyomorel** [comment]
Как и обещал отправляю промежуточный этап своего API
https://github.com/artyomorel/Bookshop/tree/develop

**Reply by pingvin1308** [reply]
Отличная реализация, не хватает только мелочей, вроде документации к апи, валидации входных параметров. Но это мелочи, по ходу работы наберешься нужно опыта. Мне нужно твое резюме)

**Comment by artyomorel** [comment]
И вот консольное приложение игры про Волка , Козу и Овцу 
https://github.com/artyomorel/WolfGoatCabbage/tree/Develop

**Reply by pingvin1308** [reply]
Думаю стоит немного переработать логику работы с игровыми объектами. Сравнение объектов используя магические числа выглядит неочевидным. Думаю стоит сделать наследников от класса GameObject, Волка, Капусту и Козу
И попробовать проверять используя типы, вместо чисел
